pkg/controller/acmechallenges/sync.go

```diff
@@ -260,19 +253,8 @@ func (c *controller) handleFinalizer(ctx context.Context, ch *cmacme.Challenge)
 	}
 
 	defer func() {
-		// call UpdateStatus first as we may have updated the challenge.status.reason field
-		ch, updateErr := c.updateStatusOrApply(ctx, ch)
-		if updateErr != nil {
-			err = utilerrors.NewAggregate([]error{err, updateErr})
-			return
-		}
 		// call Update to remove the metadata.finalizers entry
 		ch.Finalizers = ch.Finalizers[1:]
```

Comment:
Yes this would have worked pre-v1.8 as our finalizer was created together with the `Challenge` itself in the orders controller so would have always been first.

Reply:
There's nothing to stop a foreign controller re-ordering the list of finalizers though. 
 * https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-deletion